BurnOutSharp/ProtectionType/Tages.cs

```diff
@@ -20,110 +20,190 @@ public string CheckContents(string file, byte[] fileContent, bool includeDebug,
-                return protections;
-        }
-
-        /// <inheritdoc/>
-        public string CheckFilePath(string path)
-        {
-            // "Wave.aif" was a former check here, but it over-matched
-            if (Path.GetFileName(path).Equals("Tages.dll", StringComparison.OrdinalIgnoreCase))
+            // The following files are supposed to only be found inside the driver setup executables.
+            else if (Path.GetFileName(path).Equals("ithsgt.sys", StringComparison.OrdinalIgnoreCase))
             {
-                return "TAGES";
+                return "TAGES Driver";
             }
-            else if (Path.GetFileName(path).Equals("tagesclient.exe", StringComparison.OrdinalIgnoreCase))
+            else if (Path.GetFileName(path).Equals("lilsgt.sys", StringComparison.OrdinalIgnoreCase))
             {
-                return "TAGES Activation Client " + Utilities.GetFileVersion(path);
+                return "TAGES Driver";
             }
-            else if (Path.GetFileName(path).Equals("TagesSetup.exe", StringComparison.OrdinalIgnoreCase))
+            else if (Path.GetFileName(path).Equals("atksgt.sys", StringComparison.OrdinalIgnoreCase))
             {
-                return "TAGES Setup " + Utilities.GetFileVersion(path);
+                return "TAGES Driver";
             }
-            else if (Path.GetFileName(path).Equals("TagesSetup_x64.exe", StringComparison.OrdinalIgnoreCase))
+            else if (Path.GetFileName(path).Equals("lirsgt.sys", StringComparison.OrdinalIgnoreCase))
             {
-                return "TAGES Setup " + Utilities.GetFileVersion(path);
+                return "TAGES Driver";
+            }
+
+            // The following files appear to be container formats for TAGES, but little is currently known about them.
+            else if (Path.GetFileName(path).Equals("GameModule.elb", StringComparison.OrdinalIgnoreCase))
+            {
+                return "TAGES/SolidShield Game Executable Container";
+            }
+            else if (Path.GetFileName(path).Equals("InstallModule.elb", StringComparison.OrdinalIgnoreCase))
+            {
+                return "TAGES/SolidShield Installer Container";
+            }
+
+            // Not much is known about this file, but it seems to be related to what PiD reports as "protection level: Tages BASIC". Seems to always be found with other KWN files.
+            else if (Path.GetFileName(path).Equals("GAME.KWN", StringComparison.OrdinalIgnoreCase))
+            {
+                return "TAGES (BASIC?)";
             }
 
             return null;
         }
 
+        private string GetVersion(PortableExecutable pex)
+        {
+            // Check the file version first
+            string version = Utilities.GetFileVersion(pex);
+            if (!string.IsNullOrEmpty(version))
+                return version;
+
+            // Then check the manifest version
+            version = Utilities.GetManifestVersion(pex);
```

Comment:
Is there a situation where it hits the Manifest version?

Reply:
There shouldn't be, but the manifest version should also always be the same as file version for these checks, so I kept this as a backup.
![explorer_jQwj4eSijq](https://user-images.githubusercontent.com/24215969/157522602-13dd855b-73ea-4d0f-a278-2d76e4d166a3.png)